pootle/apps/pootle_store/updater.py

```diff
@@ -137,6 +137,30 @@ def should_update_target(self):
                 self.conflict_found
                 and self.update.resolve_conflict == POOTLE_WINS))
 
+    @property
+    def should_update_source(self):
+        return (
+            self.newunit
+            and self.source_updated)
+
+    @property
+    def should_update_state(self):
+        return (
+            self.newunit
+            and self.state_updated)
+
+    @cached_property
+    def source_updated(self):
+        if not self.newunit:
+            return False
+        return self.unit.source == self.newunit.source
+
+    @cached_property
+    def state_updated(self):
+        if not self.newunit:
+            return False
+        return self.unit.isfuzzy() != self.newunit.isfuzzy()
```

Comment:
i think they were factored out from existing logic - but maybe i added - ideally the logic is here rather than in the model